src/test/java/com/amazonaws/encryptionsdk/ParsedCiphertextTest.java

```diff
@@ -0,0 +1,94 @@
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.amazonaws.encryptionsdk;
+
+import com.amazonaws.encryptionsdk.internal.StaticMasterKey;
+import com.amazonaws.encryptionsdk.model.CiphertextHeaders;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.amazonaws.encryptionsdk.exception.BadCiphertextException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Arrays;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.spy;
+
+public class ParsedCiphertextTest extends CiphertextHeaders {
+    private StaticMasterKey masterKeyProvider;
+    private AwsCrypto encryptionClient_;
+
+    @Before
+    public void init() {
+        masterKeyProvider = spy(new StaticMasterKey("testmaterial"));
+
+        encryptionClient_ = new AwsCrypto();
+        encryptionClient_.setEncryptionAlgorithm(CryptoAlgorithm.ALG_AES_128_GCM_IV12_TAG16_HKDF_SHA256);
+    }
+
+    @Test()
+    public void goodParsedCiphertext() {
+        final int byteSize = 0;
+        final int frameSize = 0;
+        final byte[] plaintextBytes = new byte[byteSize];
+
+        final Map<String, String> encryptionContext = new HashMap<String, String>(1);
+        encryptionContext.put("ENC1", "ParsedCiphertext test with %d" + byteSize);
+
+        encryptionClient_.setEncryptionFrameSize(frameSize);
+
+        final byte[] cipherText = encryptionClient_.encryptData(
+                masterKeyProvider,
+                plaintextBytes,
+                encryptionContext).getResult();
+        final ParsedCiphertext pCt = new ParsedCiphertext(cipherText);
+
+        assertNotNull(pCt.getCiphertext());
+        assertTrue(pCt.getOffset() > 0);
+    }
+
+    @Test(expected = BadCiphertextException.class)
+    public void incompleteZeroByteCiphertext() {
+        final byte[] cipherText = {};
+        ParsedCiphertext pCt = new ParsedCiphertext(cipherText);
+    }
+
+    @Test(expected = BadCiphertextException.class)
+    public void incompleteSingleByteCiphertext() {
+        final byte[] cipherText = {0};
+        ParsedCiphertext pCt = new ParsedCiphertext(cipherText);
+    }
+
+    @Test(expected = BadCiphertextException.class)
+    public void incompleteCiphertext() {
+        final int byteSize = 0;
+        final int frameSize = 0;
+        final byte[] plaintextBytes = new byte[byteSize];
+
+        final Map<String, String> encryptionContext = new HashMap<String, String>(1);
+        encryptionContext.put("ENC1", "ParsedCiphertext test with %d" + byteSize);
+
+        encryptionClient_.setEncryptionFrameSize(frameSize);
+
+        final byte[] cipherText = encryptionClient_.encryptData(
+                masterKeyProvider,
+                plaintextBytes,
+                encryptionContext).getResult();
+        byte[] incompleteCipherText = Arrays.copyOfRange(cipherText, 1, cipherText.length);
```

Comment:
This isn't testing what you think because it is dropping the first byte of the header while maintaining the result of the ciphertext. This will result in a complete but invalid) header. That isn't what you are trying to test.

Try this:
```
ParsedCiphertext parsed = new ParsedCiphertext(cipherText);
byte[] incompleteCiphertext = Arrays.copyOf(parsed.getCiphertext(), parsed.getOffset() - 1);
```